### PR TITLE
feat: add whatsapp twilio provider with templates

### PR DESCRIPTION
**Pull request by LakshayaT**
### Pre-requisites

- [x] I have gone through the Contributing guidelines for [Submitting a Pull Request (PR)](../../CONTRIBUTING.md#submitting-a-pull-request-pr) and ensured that this is not a duplicate PR.
- [ ] I have performed preliminary testing using the [test suite](../../apps/api/OsmoX.postman_collection.json) to ensure that any existing features are not impacted and any new features are working as expected.
- [x] I have updated the required [api docs](../../apps/api/docs/) as applicable.
- [x] I have added/updated test cases to the [test suite](../../apps/api/OsmoX.postman_collection.json) as applicable

### PR Details

PR details have been updated as per the given format (see below)

- [x] PR title adheres to the format specified in guidelines (e.g., `feat: add admin login endpoint`)
- [x] Description has been added
- [x] Related changes have been added (optional)
- [ ] Screenshots have been added (optional)
- [x] Query request and response examples have been added (as applicable, in case added or updated)
- [x] Documentation changes have been listed (as applicable)
- [ ] Test suite output is added (as applicable)
- [ ] Pending actions have been added (optional)
- [ ] Any other additional notes have been added (optional)

### Additional Information

- [x] Appropriate label(s) have been added (`ready for review` should be added if the PR is ready to be reviewed)
- [x] Assignee(s) and reviewer(s) have been added (optional)

---

**Description:**

This PR adds support for a new WhatsApp Twilio (Business) notification for creating and sending notifications using Content Template Builder templates.

**Related changes:**

- Add wa-twilio-business module and service files, with logic for sending message
- Add waTwilioBusiness queue config
- Add waTwilioBusiness data DTO
- Update notifications constants to add WA Twilio Business channel type
- Update NotificationsQueueProducer to add case for WA Twilio Business channel type
- Add WaTwilioBusinessNotificationsConsumer for processing notifications in queue
- Update IsDataValidConstraint to add case for WA Twilio Business channel type
- Update NotificationsModule to add the WaTwilioBusiness module, queue and consumer
- Update migration seed data to add master_provider details for WA_TWILIO_BUSINESS
- Update postman collection to add requests and test cases for provider

**Screenshots:**

N/A

**Query request and response:**

Sample request and cURL:

```bash
curl --location 'localhost:3000/notifications' \
--header 'Content-Type: application/json' \
--header 'Authorization: Bearer OsmoX-test-key' \
--data '{
  
  "providerId": 7,
  "data": {
    "contentSid": "HXxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
    "from": "MGxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
    "contentVariables": {
      "1": "Lakshaya",
      "2": "2",
      "3": "29-04-2024",
      "4": "03-05-2024",
      "5": "Sunday, 05-05-2024"
    },
    "to": "+91xxxxxxxxxx"
  }
}'
```

```json
{
  "providerId": 7,
  "data": {
    "contentSid": "HXxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
    "from": "MGxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
    "contentVariables": {
      "1": "Lakshaya",
      "2": "2",
      "3": "29-04-2024",
      "4": "03-05-2024",
      "5": "Sunday, 05-05-2024"
    },
    "to": "+91xxxxxxxxxx"
  }
}
```

Response:

```json
{
  "status": "success",
  "data": {
    "notification": {
      "providerId": 7,
      "data": {
        "contentSid": "HXxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
        "from": "MGxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
        "contentVariables": {
          "1": "Lakshaya",
          "2": "2",
          "3": "29-04-2024",
          "4": "03-05-2024",
          "5": "Sunday, 05-05-2024"
        },
        "to": "+91xxxxxxxxxx"
      },
      "applicationId": 1,
      "createdBy": "sampleOsmoXApp",
      "updatedBy": "sampleOsmoXApp",
      "result": null,
      "id": 120,
      "deliveryStatus": 1,
      "createdOn": "2024-05-01T07:23:51.000Z",
      "updatedOn": "2024-05-01T07:23:51.000Z",
      "status": 1
    }
  }
}
```

**Documentation changes:**

- Add the documentation for wa-Twilio-Business usage
- Update usage-guide to add link to this document

**Test suite output:**

N/A

**Pending actions:**

N/A

**Additional notes:**

N/A

## Summary by CodeRabbit

- **New Features**
  - Added support for Twilio WhatsApp (Business) notifications, enabling users to send and receive messages via this new channel.
  - Introduced new documentation and usage guides for Twilio WhatsApp Business integration.

- **Bug Fixes**
  - Updated Postman collection IDs for improved tracking and consistency.

- **Documentation**
  - Added new guides for setting up and using the Twilio WhatsApp Business platform.

- **Refactor**
  - Implemented validation logic for the Twilio WhatsApp Business data in the API.

- **Tests**
  - Introduced tests for the new Twilio WhatsApp Business service functionality.

- **Chores**
  - Updated database seeding scripts to include new provider types and configurations.

### CODE REVIEW

File: apps/api/src/modules/providers/wa-twilio-business/wa-twilio-business.service.spec.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+      providers: [WaTwilioBusinessService],
+    }).compile();
+
+    service = module.get<WaTwilioBusinessService>(WaTwilioBusinessService);
+  });
+
+  it('should be defined', () => {
+    expect(service).toBeDefined();
+  });
+});
```
[the remark this answers is not in the document]
Comment (LakshayaT) [reply apps/api/src/modules/providers/wa-twilio-business/wa-twilio-business.service.spec.ts:18]: We have test cases added in postman collection, so this much should be good for now

File: apps/api/docs/channels/wa-Twilio-Business.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+      "2": "52" // Numbers should also be strings
+    },
+    "to": "+919004812051"
+  }
+}
+```
+
+References
+
+- https://www.twilio.com/docs/content/send-templates-created-with-the-content-template-builder
```
[the remark this answers is not in the document]
Comment (LakshayaT) [reply apps/api/docs/channels/wa-Twilio-Business.md]: Addressed in https://github.com/OsmosysSoftware/osmo-x/pull/229/commits/cd654505fcd4efe06ddbad693c199135ced41168

File: apps/api/docs/channels/wa-Twilio-Business.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
+## Twilio WhatsApp (Business)
+
+Using the WhatsApp Business Platform with Twilio, you can send and receive messages to WhatsApp users using the same Twilio Messaging APIs you already know and enjoy. Dive into the Twilio SDKs and helper libraries, see the quickstart and API reference docs, read through guides on templates and Twilio phone numbers, and find the sample code you'll need.
+
+This particular provider can be used to send WhatsApp messages generated using Content Template Builder.
+
+### Values to Update in Database
+
+When using Twilio to send whatsApp messages via their API/Client, you need to provide certain variables that hold the Twilio configuration details. Here are the values you need to update in table `notify_providers`:
```
[the remark this answers is not in the document]
Comment (LakshayaT) [reply apps/api/docs/channels/wa-Twilio-Business.md]: Fixed in https://github.com/OsmosysSoftware/osmo-x/pull/229/commits/9e62903d3994180e4768adbf30bd1831ce072c56

File: apps/api/docs/channels/wa-Twilio-Business.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+Set field `is_enabled` as 1.
+
+Then set the following configurations in the `configuration` field:
+
+| Key                   | Description            |
+| --------------------- | ---------------------- |
+| TWILIO_WA_ACCOUNT_SID | Twilio SMS account SID |
+| TWILIO_WA_AUTH_TOKEN  | Twilio SMS auth token  |
+
```
[the remark this answers is not in the document]
Comment (LakshayaT) [reply apps/api/docs/channels/wa-Twilio-Business.md:19]: This is auto-formatted and provides more readability in the raw markdown text.

**Comment by LakshayaT** [comment]
@coderabbitai review

File: apps/api/docs/channels/wa-Twilio-Business.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
+## Twilio WhatsApp (Business)
+
+Using the WhatsApp Business Platform with Twilio, you can send and receive messages to WhatsApp users using the same Twilio Messaging APIs you already know and enjoy. Dive into the Twilio SDKs and helper libraries, see the quickstart and API reference docs, read through guides on templates and Twilio phone numbers, and find the sample code you'll need.
```
[the remark this answers is not in the document]
Comment (LakshayaT) [reply apps/api/docs/channels/wa-Twilio-Business.md:3]: The spelling is correct already.

File: apps/api/docs/channels/wa-Twilio-Business.md [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+{
+  // Set your respective providerId
+  "providerId": 7,
+  "channelType": 7,
+  "data": {
+    "contentSid": "HXXXXXXXXX",
+    "from": "MGXXXXXXXX",
+    "contentVariables": {
+      "1": "Name",
+      "2": "52" // Numbers should also be strings
```
[the remark this answers is not in the document]
Comment (LakshayaT) [reply apps/api/docs/channels/wa-Twilio-Business.md:42]: I think this should be clear enough for the time being. Can update it if really needed later

**Review by osm-vishnukyatannawar — APPROVED** [review]

**Pull request merged**